aces/joint_deconvolution/feather_all_spws.py

```diff
@@ -0,0 +1,260 @@
+                    imreframe(
+                        imagename=tp_7m_cube,
+                        restfreq=twelve_m_freq['value'] + ' Hz',
+                        output=tp_7m_cube + '.reframe'
+                    )
+                # If the feathered TP+7m+12m cube does not exist, feather the 12m and TP+7m cubes together
+                try:
+                    feather(
+                        imagename=str(OUTPUT_TP_7M_12M),
+                        highres=twelve_m_cube_im + '.commonbeam' if os.path.isdir(twelve_m_cube_im + '.commonbeam') else twelve_m_cube_im,
+                        lowres=tp_7m_cube + '.reframe' if (Path(tp_7m_cube) / '.reframe').is_dir() else tp_7m_cube
+                    )
+                except Exception as e:
+                    print(f"Feather failed with highres={twelve_m_cube_im}: {e}")
+                    try:
+                        feather(
+                            imagename=str(OUTPUT_TP_7M_12M),
+                            highres=twelve_m_cube_im + '.commonbeam',
+                            lowres=tp_7m_cube + '.reframe' if (Path(tp_7m_cube) / '.reframe').is_dir() else tp_7m_cube
+                        )
+                    except Exception as e:
+                        print(f"Feather task failed again with 12M data smoothed to a common beam. Skipping feathering: {e}")
+
+        if OUTPUT_TP_7M_12M.is_dir() and not os.path.isfile(str(OUTPUT_TP_7M_12M) + '.fits'):
+            # Export the feathered cubes and the 12m weights to FITS files
+            export_fits(
+                imagename=str(OUTPUT_TP_7M_12M),
+                fitsimage=str(OUTPUT_TP_7M_12M) + '.fits'
+            )
+    else:
+        print(f"One or more cubes do not exist for observation Sgr_A_st_{obs_id}. Skipping this one ...")
+
+
+def create_feathercubes(ACES_WORKDIR, ACES_DATA, ACES_ROOTDIR, line_spws, MOLECULE, process_12M=True):
+    """
+    Loop over each SB and identify the relevant MOUS IDs
+    Retrieves the corresponding data cubes and feathers them
+
+    The function can optionally include 12m data
+
+    Parameters
+    ----------
+    ACES_WORKDIR : Path
+        Path object representing the working directory.
+    ACES_DATA : Path
+        Path object representing the directory where the data files are stored.
+    ACES_ROOTDIR : Path
+        Path object representing the root directory of ACES.
+    line_spws : dict
+        Dictionary mapping molecule identifiers to spectral window information.
+    MOLECULE : str
+        The molecule being processed (e.g. 'HNCO').
+    process_12M : bool, optional
+        If True, 12m data is processed and included in the feathering process. Default is True.
+    """
+    # Load the SB information
+    sb_names = pd.read_csv(ACES_ROOTDIR / 'aces/data/tables/aces_SB_uids.csv')
+
+    # Define common parts of the file naming scheme
+    generic_name = '.Sgr_A_star_sci.spw'
+    prefix = 'member.uid___A001_'
+    regions = ['ap']
```

Comment:
is this intentionally hard-coded?  Seems like it would make more sense as a parameter?